third_party/move/move-compiler-v2/tests/bytecode-generator/wildcard7.exp

```diff
@@ -42,6 +42,14 @@ public fun m::test(): u8 {
 }
 
 
+Diagnostics:
+warning: Unused assignment to `q`. Consider removing or prefixing with an underscore: `_q`
```

Comment:
This warning is redundant.  We could suppress in the case of no uses, in favor of "unused variable".  But it's simpler to remove the declaration when the "unused variable" warning is generated.  Filed https://github.com/aptos-labs/aptos-core/issues/13791 to track that.

Reply:
I explored removing the unused declaration back in the AST and it's pretty messy.  Would be simpler to suppress the redundant warning here in the case of a variable declaration.  Perhaps that will require a change to the bytecode to track whether an assignment/loc corresponds to a variable declaration.